src/library/dao/trackdao.cpp

```diff
@@ -1293,6 +1303,38 @@ TrackPointer TrackDAO::getTrackFromDB(TrackId trackId) const {
         }
     }
 
+    // Data migration: Reload track total from file tags if not initialized
+    // yet. The added column "tracktotal" has been initialized with the
+    // default value "//".
+    // See also: Schema revision 26 in schema.xml
+    if (pTrack->getTrackTotal() == "//") {
+        // Just in case: Use track total from the track number itself if
+        // it can be split into two parts and the second part os not empty.
+        const std::pair<QString, QString> splittedTrackNumber(
+                TrackNumbers::splitString(pTrack->getTrackNumber()));
```

Comment:
I do not like std::pair because of its anonymous members. 
I would use output parameters instead.